Sources/OktaAnalytics/OktaAnalytics.swift

```diff
@@ -10,17 +10,49 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 import Foundation
+import Combine
 import OktaLogger
+import OktaSQLiteStorage
+import GRDB
 
 /**
  Analytics class for holding and tracking for different tracking provider
  */
+
 @objc
 public final class OktaAnalytics: NSObject {
 
     private static var providers = [String: AnalyticsProviderProtocol]()
     private static var lock = ReadWriteLock()
 
+    private static var storage: AnalyticsStorage = {
+        let logger = OktaLogger()
+        logger.addDestination(
+            OktaLoggerConsoleLogger(
+                identifier: "com.okta.Analytics.storage",
+                level: OktaLoggerLogLevel.debug,
+                defaultProperties: nil
+            )
+        )
+        let storage = AnalyticsStorage(logger: logger)
+        lock.writeLock()
+        Task(priority: .high) {
+            do {
+                try await storage.initializeDB(forSecurityApplicationGroupIdentifier: securityAppGroupIdentifier)
+                lock.unlock()
+            } catch {
+                lock.unlock()
+                logger.log(error: error as NSError)
+            }
+        }
+        return storage
+    }()
+
+    public static var securityAppGroupIdentifier = "" {
+        didSet {
+            _ = storage
```

Comment:
Is it for triggering storage creation and initialize? :)

Reply:
updated with new change